### PR TITLE
Added a simple controller skeleton

### PR DESCRIPTION
**Pull request by ellistarn**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/controllers/provisioner/v1alpha1/controller.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+// Reconcile executes a control loop for the resource
+func (c *Controller) Reconcile(object controllers.Object) error {
+	_ = object.(*v1alpha1.Provisioner)
+
+	// 1. List Pods where pod.spec.nodeName = ''
+	pods := &v1.PodList{}
+	if err := c.Client.List(context.Background(), pods, client.MatchingFields{"spec.nodeName": ""}); err != nil {
+		return fmt.Errorf("Listing unscheduled pods, %w", err)
+	}
```
Comment (prateekgogia) [inline pkg/controllers/provisioner/v1alpha1/controller.go:55]: If we get pending pods this way, could we be getting pods which scheduler is yet to schedule and there is enough capacity for them.

**Review by prateekgogia — CHANGES_REQUESTED** [review]

File: pkg/apis/autoscaling/v1alpha1/zz_generated.deepcopy.go [hunk 2]
```diff
@@ -368,9 +368,9 @@ func (in *MetricsProducerSpec) DeepCopyInto(out *MetricsProducerSpec) {
 		*out = new(ReservedCapacitySpec)
 		(*in).DeepCopyInto(*out)
 	}
-	if in.ScheduledCapacity != nil {
-		in, out := &in.ScheduledCapacity, &out.ScheduledCapacity
-		*out = new(ScheduledCapacitySpec)
+	if in.Schedule != nil {
+		in, out := &in.Schedule, &out.Schedule
+		*out = new(ScheduleSpec)
```
[the remark this answers is not in the document]
Comment (ellistarn) [reply pkg/apis/autoscaling/v1alpha1/zz_generated.deepcopy.go:373]: Not sure what you mean.

File: pkg/controllers/provisioner/v1alpha1/controller.go (hunk 1 above)
Comment (ellistarn) [reply pkg/controllers/provisioner/v1alpha1/controller.go:55]: We probably need to filter by the status condition as well.

Comment (ellistarn) [reply pkg/controllers/provisioner/v1alpha1/controller.go:55]: TODO :)

File: pkg/apis/autoscaling/v1alpha1/zz_generated.deepcopy.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (prateekgogia) [reply pkg/apis/autoscaling/v1alpha1/zz_generated.deepcopy.go:373]: This generated code is for scheduled capacity changes which @njtran did, I think we missed adding this in main branch.

**Review by prateekgogia — APPROVED** [review]

**Pull request merged**